File: src/main/kotlin/application/controller/dialogs/rule/SharedWidgets.java
package application.controller.dialogs.rule;

import application.controller.NeighbourhoodSelector;

public class SharedWidgets {
    private static final NeighbourhoodSelector neighbourhoodSelector = new NeighbourhoodSelector(2);

    public static NeighbourhoodSelector getNeighbourhoodSelector() {
        NeighbourhoodSelector newNeighbourhoodSelector = new NeighbourhoodSelector(2);

        // Bidirectional Binding
        neighbourhoodSelector.getSpinnerRange().valueProperty().addListener((obs, oldValue, newValue) ->
                newNeighbourhoodSelector.getSpinnerRange().getValueFactory().setValue(newValue));
        newNeighbourhoodSelector.getSpinnerRange().valueProperty().addListener((obs, oldValue, newValue) ->
                neighbourhoodSelector.getSpinnerRange().getValueFactory().setValue(newValue));

        //neighbourhoodSelector.setOnWeightsChanged(() ->
        //        newNeighbourhoodSelector.setWeights(neighbourhoodSelector.getRawWeights()));
        //newNeighbourhoodSelector.setOnWeightsChanged(() ->
        //        neighbourhoodSelector.setWeights(newNeighbourhoodSelector.getRawWeights()));

        return newNeighbourhoodSelector;
    }
}

File: src/main/kotlin/application/controller/dialogs/rule/SharedWidgets.kt
package application.controller.dialogs.rule

import application.controller.NeighbourhoodSelector
import javafx.beans.value.ObservableValue

object SharedWidgets {
    val neighbourhoodSelector = NeighbourhoodSelector(2)
        get() {
            val newNeighbourhoodSelector = NeighbourhoodSelector(2)

            // Bidirectional Binding
            field.spinnerRange.valueProperty()
                .addListener { _: ObservableValue<out Int>, _: Int, newValue: Int ->
                    newNeighbourhoodSelector.spinnerRange.valueFactory.setValue(newValue)
                }
            newNeighbourhoodSelector.spinnerRange.valueProperty()
                .addListener { _: ObservableValue<out Int>, _: Int, newValue: Int ->
                    field.spinnerRange.valueFactory.setValue(newValue)
                }

            return newNeighbourhoodSelector
        }
}